compiler/src/toEC.ml

```diff
@@ -988,14 +989,14 @@ module Leak = struct
     | Pload (ws,x,e) -> 
       is_init env x (Valid (ws, snd (add_ptr pd (gkvar x) e)) :: safe_e_rec pd env safe e)
     | Papp1 (_, e) -> safe_e_rec pd env safe e
-    | Pget (aa, ws, x, e) -> 
+    | Pget (aa, ws, x, e) ->
       assert (aa = Warray_.AAscale); (* NOT IMPLEMENTED *)
       let x = x.gv in
       let safe = 
         let (_s,option) = Mv.find (L.unloc x) env.vars in
         if option then Initai(ws, L.unloc x, e) :: safe 
         else safe in
-      in_bound ws x e :: safe 
+      in_bound x e :: safe
```

Comment:
I don't understand why the ws argument of in bound is removed.
stack u64[1] t;
x = t[u16 2];
I have the impression that this code will be rejected but it should not.

Reply:
Let me drop the first commit that is unrelated and clearly bad.